### ISSUE TITLE
Catch2 XML reporter format change breaks Test Adapter

### ISSUE THREAD

**Issue by PaulFocus** [JohnnyHendriks/TestAdapter_Catch2#52]
We switched from an old Catch2 developer version labeled in **catch_amalgated.hpp** as
```
Generated: 2020-10-08 13:59:26.616931
```
to [git commit 4436a60](https://github.com/catchorg/Catch2/commit/4436a604566f18091489b55af215b8d32afe657a), which is quite recent.

This unfortunately broke this Test Adapter, because the XML output structure, which is processed by this adapter, changed: test group has been removed and element "Catch" renamed to "Catch2TestRun" in [Catch2 commit e5938007f7dab5c9886ca561d09106ee4d7f2301](https://github.com/catchorg/Catch2/commit/e5938007f7dab5c9886ca561d09106ee4d7f2301) of September 7th this year.

Both Catch2 commits generate the same version string when calling the executable with argument ``--libidentify``:
```
description:    A Catch2 test executable
category:       testframework
framework:      Catch Test
version:        3.0.0-preview.3
```

Therefore it is not possible for the Test Adapter yet to distinguish between new and old XML output and parse the results accordingly.

We attached the different outputs for one of our test projects. It only contains a single test case. The output has been generated by:
```
test_d *narrowcast* -s -r xml -o output.xml
```

Best regards,
Paul

[xml-output-old-and-new.zip](https://github.com/JohnnyHendriks/TestAdapter_Catch2/files/7352370/xml-output-old-and-new.zip)

**Comment by JohnnyHendriks** [comment]
Thank you for the heads up. I'm using the v3.0.0-preview3 release myself, which apparently does not have these changes yet as everything seems to work fine for me at the moment. However, I'm not keeping track of the latest unreleased changes as I consider them too volatile. As soon as a new official preview version is available (or when version 3 gets released, whichever comes first) I will have a look at what changes are required to keep things working.

**Comment by PaulFocus** [comment]
The preview version has been increased to v4 in this commit:
https://github.com/catchorg/Catch2/commit/f4af9f69265d009a457aa99d1075cfba78652a66

**Comment by JohnnyHendriks** [comment]
Thank you, I'm already aware. Already updated code locally, and am currently working on updating the unittests to make sure everything works as expected.

**Comment by JohnnyHendriks** [comment]
Fixed in v1.8.0 release.
